game/script-ch30.rpy

```diff
@@ -422,3 +416,106 @@ label extras_menu:
+                    n 1tllss "Now,{w=0.1} where was I...?{w=1}{nw}"
+                    extend 1unmbo " Oh,{w=0.1} right.{w=1}{nw}"
+
+                elif jn_affinity.get_affinity_state() >= jn_affinity.NORMAL:
+                    n 1flleml "G-{w=0.1}good!{w=1}{nw}"
+                    extend 1kllpol " Good...{w=1}{nw}"
+                    n 1tslpu "Now...{w=0.3} what was I saying again?{w=0.5}{nw}"
+                    extend 1nnmbo " Oh,{w=0.1} right.{w=1}{nw}"
+
+                elif jn_affinity.get_affinity_state() >= jn_affinity.DISTRESSED:
+                    n 1fsqfr "...Thank you.{w=1}{nw}"
+                    n 1fslpu "As I was {i}saying{/i}...{w=1}{nw}"
+
+                else:
+                    n 1fcsfr "Whatever.{w=1}{nw}"
+                    n 1fsqsl "{cps=/2}As I was saying.{/cps}{w=1}{nw}"
+
+                return
+
+            # Continue force quit
+            "...":
+                hide screen hkb_overlay
+                if jn_affinity.get_affinity_state() >= jn_affinity.AFFECTIONATE:
+                    n 1kwmem "Come on,{w=0.2} [player]...{w=1}{nw}"
+                    play audio glitch_c
+                    stop music
+                    n 1kcsup "...!{nw}"
+
+                elif jn_affinity.get_affinity_state() >= jn_affinity.NORMAL:
+                    n 1fwmun "...Really,{w=0.2} [player]?{w=1}{nw}"
+                    play audio glitch_c
+                    stop music
+                    n 1kcsfu "Hnnng-!{nw}"
+
+                elif jn_affinity.get_affinity_state() >= jn_affinity.DISTRESSED:
+                    n 1fslun "Don't let the door hit you on the way out.{w=1}{nw}" 
+                    extend 1fsqem " Jerk.{w=1}{nw}"
+                    play audio glitch_c
+                    stop music
+                    n 1fcsan "Nnngg-!{nw}"
+
+                else:
+                    n 1fslun "Heh.{w=1}{nw}"
+                    extend 1fsqfr "...Maybe you {i}shouldn't{/i} come back.{w=1}{nw}"
+                    play audio glitch_c
+                    stop music
+                    n 1fcsfr "...{nw}"
+
+                    if (random.randint(0, 10) == 1):
+                        play sound glitch_d loop
+                        show glitch_garbled_red zorder 99 with vpunch
+                        $ renpy.pause(random.randint(4,13))
+                        stop sound
+                        play audio glitch_e
+                        show glitch_garbled_n zorder 99 with hpunch
+                        $ renpy.pause(0.025)
+                        hide glitch_garbled_n
+                        hide glitch_garbled_red
+
+                # Apply consequences for force quitting, then glitch quit out
+                $ jn_relationship("affinity-")
+                $ jn_apologies.add_new_pending_apology(jn_apologies.TYPE_SUDDEN_LEAVE)
+                if jn_globals.player_is_ingame:
+                    $ jn_apologies.add_new_pending_apology(jn_apologies.JNApologyTypes.cheated_game)
```

Comment:
I don't think this is intented. We shouldn't accuse the player of cheating for not saying goodbye

Reply:
This is a tricky one; I originally added it under the line of thought that since the player is given a lot of warnings regarding force quit, that to force quit during a game is like refusing to admit defeat, or being unsportsmanlike/sour.

This wouldn't apply when the player is forced to quit through an error of some kind, but I'll remove for now and potentially look at making a new apology type specifically for quitting during a game.